app/controllers/bulk_updates_controller.rb

```diff
@@ -54,14 +56,15 @@ def new
 
   def preview
     add_breadcrumb 'Preview', bulk_update_preview_path
-
-    @task = construct_task(params)
+    permitted = params.to_unsafe_h unless params.nil?# need to understand what this is doing more, think related to nested parameters not permitted.
```

Comment:
do we have a TODO syntax for these comments? Would this be relevant here?

Reply:
We were using TODO RAILS5 for awhile, but all of those have been removed.   There are a few RAILS5 comments left around, such as those around the "datetime" changes.

Reply:
I'm seeing comments like this in quite a few places (not having gone through everything yet). Some use `to_unsafe_h`, and some use the `safe_hash` method above. I'm wondering if there is a difference. Also, these should be addressed by MMT-2030, right?

Reply:
This is a difficult one.  safe_hash and to_unsafe_h work differently,  but I suspect that only safe_hash should be used in our code.   Unfortunately,  to_unsafe_h is part of rails so we can't remove it. 

I recommend that we leave these as they are and I will add some notes to MMT-2030 about the issues.  Then we can decide if/when/how to fix them.  

Note: I had thought that MMT-2030 was OBE, since we fixed the permit issues as part of the general bug fixing.

Reply:
safe_hash is defined in the application controller.  I believe that it takes a key, ensures that the key is there and then mass permits the keys.  to_unsafe_h just makes a hash of the keys.

Reply:
I think that plan makes sense. But in my opinion 2030 is not OBE, because I think we need to actually name all the params we want to permit as Rails is trying to force us to do.